### ISSUE TITLE
maybe [] f -> foldMap f

### ISSUE THREAD

**Issue by googleson78** [ndmitchell/hlint#1157]
Would such a hint be useful? Notably, it might possibly not be correct in the presence of `OverloadedLists`

**Comment by ndmitchell** [comment]
Seems a bit confusing to me - its only slightly shorter, but I wouldn't have guessed what it does.

**Comment by googleson78** [comment]
Maybe it's just a thing at the place I work at! To me, it feels in the same line as
```haskell
case mx of
  Nothing -> pure Nothing
  Just x -> Just <$> ...
-- ->
for mx \x -> ...
```

Maybe it can be added to the same (not-enabled by default) "generalise for conciseness" group? (this being motivated because the group has e.g. `either mempty -> foldMap`)

**Comment by ndmitchell** [comment]
Yep, no objection to generalise for conciseness. I think the difference between those two is that the complexity of a case statement is high, while the complexity of `maybe []` is low.

**Comment by yairchu** [comment]
Not that there's already a similar rule [in the `generalise-for-conciseness` group](https://github.com/ndmitchell/hlint/blob/master/data/hlint.yaml#L866): `- warn: {lhs: maybe mempty, rhs: foldMap}`

And if not considered more concise, then this rule might fit in the `generalise` group

**Comment by yairchu** [comment]
Can be closed now? (exists as the default-off group "generalise")

**Comment by ndmitchell** [comment]
Yep, thanks for the patch!
